mpmp/utilities/tcga_utilities.py

```diff
@@ -829,4 +829,74 @@ def compress_and_save_data(data_type,
     return transformed_data_df
 
 
+def drop_target_from_data(data_df, target_gene, gene_features):
+    """Drop target gene from feature set."""
+    symbol_map, update_map = get_symbol_map()
+
+    try:
+        target_feature = symbol_map[target_gene]
+        if target_feature in update_map.keys():
+            target_feature = update_map[target_gene]
+        target_feature = str(target_feature)
+    except KeyError:
+        # if not found in symbol_map, just use raw feature
+        target_feature = str(target_gene)
+
+    # update data by removing target
+    data_df_updated = data_df.drop(columns=target_feature)
+
+    # update gene features array
+    target_ix = data_df.columns.get_loc(target_feature)
+    gene_features_updated = np.delete(gene_features, target_ix)
+
+    return data_df_updated, gene_features_updated
+
+
+def only_target_from_data(data_df, target_gene, gene_features):
+    """Reduce feature set to only target gene."""
+    symbol_map, update_map = get_symbol_map()
+
+    try:
+        target_feature = symbol_map[target_gene]
+        if target_feature in update_map.keys():
+            target_feature = update_map[target_gene]
+        target_feature = str(target_feature)
+    except KeyError:
+        # if not found in symbol_map, just use raw feature
+        target_feature = str(target_gene)
+
+    # update data by removing all but target
+    non_gene_df = data_df.loc[:, ~gene_features]
+    gene_df = data_df.loc[:, target_feature]
+    data_df_updated = pd.concat((gene_df, non_gene_df), axis='columns')
+
+    # update gene features array, there's only one gene feature now
+    gene_features_updated = np.array(
+        [True] + ([False] * non_gene_df.shape[1])
+    )
+
+    return data_df_updated, gene_features_updated
+
+
+def get_symbol_map():
+
+    # TODO move these urls to config
```

Comment:
Is this still in the cards, or is it alright for the URL to live here?

Reply:
Still in the cards! Good catch, will do this now.